flowzone.yml

```diff
@@ -1560,7 +1560,8 @@ jobs:
         with:
           python-version: ${{ env.SUPPORTED_VERSIONS }}
 
-      - *setupPoetry
+      - <<: *setupPoetry
+        if: steps.python_poetry.outputs.enabled == 'true'
```

Comment:
@klutchell why not just add it to the anchor? Assuming setup-python is always a dependency (I can't think of a reason why it wouldn't be):

```
  - &setupPoetry # https://github.com/abatilo/actions-poetry
    name: Setup poetry
    uses: abatilo/actions-poetry@192395c0d10c082a7c62294ab5d9a9de40e48974 # v2.3.0
    if: steps.python_poetry.outputs.enabled == 'true'
    with:
      poetry-version: "1.5.1"
```

Reply:
because this step `steps.python_poetry.outputs.enabled` only runs during `is_python` and doesn't exist for the other jobs that call setupPoetry

Reply:
based on the changes from @pranasziaukas , a poetry file is required for all of the downstream python jobs, it's only during `is_python` that we need to check if we found a poetry file or not

Reply:
earlier comments withdrawn, that would be a better approach